sheerwater_benchmarking/utils/caching.py

```diff
@@ -28,6 +30,130 @@
 CHUNK_SIZE_UPPER_LIMIT_MB = 300
 CHUNK_SIZE_LOWER_LIMIT_MB = 30
 
+# Global variables for caching configuration
+global_recompute = None
+global_cache_local = None
+global_force_overwrite = None
+
+# Remote dir for all caches, except for postgres and terracotta
+REMOTE_CACHE_ROOT_DIR = "gs://sheerwater-datalake/caches/"
+# Local dir for all caches, except for postgres and terracotta
+LOCAL_CACHE_ROOT_DIR = os.path.expanduser("~/.cache/sheerwater/caches/")
+
+
+def set_global_cache_variables(recompute=None, cache_local=None, force_overwrite=None):
+    """Reset all global variables to defaults and set the new values."""
+    global global_recompute, global_cache_local, global_force_overwrite
+    global_cache_local = cache_local
+    global_force_overwrite = force_overwrite
+
+    # Handle recompute logic, different for top level and nested functions
+    if recompute:
+        global_recompute = False
+    elif isinstance(recompute, str) and recompute != 'all':
+        global_recompute = [recompute]
+    else:
+        global_recompute = recompute  # if recompute is false or a list
+
+
+def check_if_nested_fn():
+    """Check if the current scope downstream from another cached function."""
+    # Get the current frame
+    stack = inspect.stack()
+    # skip the first two frames (this function and the current cacheable function)
+    for frame_info in stack[2:]:
```

Comment:
Do we need to go farther up the stack frame now that this is itself in a function?

Reply:
Yes, I changed from 1 to 2